### PR TITLE
Read mode: reply_to relies on draft sheaf's parent

### PR DESCRIPTION
**Pull request by rtshkmr**
1 DONE change context init routines – reply to context is determined by the draft sheaf ═══════════════════════════════════════════════════════════════════════════════════════

  • CLOSING NOTE [2024-10-26 Sat 15:12]
    Done, some cleanups left though
  There’s greater clarity on how we should be setting up the reply to
  context.

1.1 Purpose of Active bool field in sheaf table
───────────────────────────────────────────────

  the purpose of the active bool on sheafs is so that we can do
  archiving of sheafs without having side-effects on other interactions
  that have spun-off because of that sheaf. This means that:
  ‣ that means people who marked your sheaf aren’t affected
  ‣ that means people who replied to your sheaf aren’t affected

  hence we should not rely on the state of published sheafs when
  determining what the reply to context should be.

1.2 Using the draft sheaf
─────────────────────────

  We can use the draft sheaf that we init for the drafting context and
  use its parent. There will be 2 cases:
  1. parent is nil ==> then the reply to is nil and we are about to create a root sheaf
  2. parent is set ==> then init that as the reply_to sheaf

### CODE REVIEW

File: lib/vyasa_web/components/contexts/read.ex [hunk 1]
```diff
@@ -198,18 +199,15 @@ defmodule VyasaWeb.Context.Read do
...
 
   There are two main ways this reply to context can be set (implicitly):
   1) because it has been set in another mode (discuss mode) ==> the db is used to mediate this.
-      => init_reply_to_context() reads from DB
-  2) via url params, as permalinked ==> this will take precendence over the db-based determining of reply_to_context
+      => init_reply_to_context() reads from DB.
+      By looking at the currently active draft sheaf, check it's parent id:
+      1A: the parent is nil: so no reply_to sheaf
+      1B: non-nil then load that parent sheaf as reply_to
+  2) TODO  @ks0m1c via url params, as permalinked ==> this will take precendence over the db-based determining of reply_to_context
```
Comment (ks0m1c) [inline lib/vyasa_web/components/contexts/read.ex:210]: if its a binding to sheaf across both modes discuss and read

Comment (ks0m1c) [reply lib/vyasa_web/components/contexts/read.ex:210]: we shouldnt override -> allow user for conscious reply_to actions

tldr allow for binding but no inference of user intent towards reply to

kiv, pop stash of current drafting table context

File: lib/vyasa_web/components/contexts/read.ex [hunk 2]
```diff
@@ -218,31 +216,31 @@ defmodule VyasaWeb.Context.Read do
...
-  - it's not ALWAYS a replying to
-
   """
   def init_reply_to_context(
         %Socket{
           assigns: %{
-            session: %{sangh: %{id: sangh_id}}
+            session: %{sangh: %{id: _sangh_id}},
+            draft_reflector: %Sheaf{
+              parent_id: parent_id
```
Comment (ks0m1c) [inline lib/vyasa_web/components/contexts/read.ex:225]: guard clauses work here

is_binary and _

**Review by ks0m1c — APPROVED** [review]

File: lib/vyasa_web/components/contexts/read.ex (hunk 2 above)
Comment (rtshkmr) [reply lib/vyasa_web/components/contexts/read.ex:225]: i dodo-ed and missed this before merging, will put this in my upcoming PR
